### DISCUSSION TITLE
Questions about VMware Cloud Director 36.0 API

### DISCUSSION THREAD

**Discussion by chenchunwei24k** [ccouzens/vcloud-rest-openapi#13] [Q&A] [answered]
Hi @ccouzens 

Sorry for bothering you!
Currently, I updated my VMware Cloud Director 10.3 to 10.4.
VMware Cloud Director 10.4 only supports API version 36.0 or greater version, other lower versions will be deprecated in the next version.
The API format between the 36.0 API and the lower version's API is totally different as below.

- Version 36.0 : https://<api_host>/cloudapi/1.0.0/
- Lower Version : https://<api_host>/api/

I'm trying to re-modify my old APIs to the new APIs format.
But I can't find the related APIs I needed in [VMware Cloud Director OpenAPI Document](https://developer.vmware.com/apis/vmware-cloud-director/v36.0/).

Searching for some related information and Google brings me here.
May I ask some questions about the VMware Cloud Director API?

- I found that this repository has recorded the API information from version 27.0 to 37.2.
Version 36.0 files information of API's format seems as same as the lower version APIs.
Why is the information on version 36.0 APIs is same as the lower version APIs? Is it means that the old API format(https://<api_host>/api/) is still useable?

- Do you know that the 36.0 APIs can execute with different query?
Such as https://<api_host>/api/query?type=vApp in the old version.

- As you know, are there any APIs that can list all the vApps information in version 36.0 APIs? or are there any operations to access the vApps information by APIs.

Sorry for bothering you, again.
And I very much appreciate your help.

Thank you!

Best Regards,
Chen

**Comment by ccouzens** [comment, marked as answer]
Hi @chenchunwei24k 

I no longer work for a company that uses VMware Cloud Director, but I should still be able to answer your questions.

>  The API format between the 36.0 API and the lower version's API is totally different as below.
>
>    Version 36.0 : https://<api_host>/cloudapi/1.0.0/
>    Lower Version : https://<api_host>/api/

There are 2 APIs.

`https://<api_host>/api/` They call this one [Cloud Director API](https://developer.vmware.com/apis/1260/vmware-cloud-director). I will call it the old API.
`https://<api_host>/cloudapi/1.0.0/` They call this one [Cloud Director OpenAPI](https://developer.vmware.com/apis/vmware-cloud-director/v37.0/). I will call it the new API.

When I used VMware Cloud Director almost everything useful had to be done with the old API. Each version they keep adding new features to the new API.

I suspect that even in version 10.4 most things are only possible with the old API.

> I'm trying to re-modify my old APIs to the new APIs format.
> But I can't find the related APIs I needed in [VMware Cloud Director OpenAPI Document](https://developer.vmware.com/apis/vmware-cloud-director/v36.0/).

I would look to see if they're available in the ["old" APIs](https://developer.vmware.com/apis/1260/vmware-cloud-director). The old style APIs are very much supported by VMware.

> I found that this repository has recorded the API information from version 27.0 to 37.2.
Version 36.0 files information of API's format seems as same as the lower version APIs.
Why is the information on version 36.0 APIs is same as the lower version APIs? Is it means that the old API format(https://<api_host>/api/) is still useable?

They are not identical, but the core APIs are quite stable and don't change much. The big difference is you'll need to send a different version in your `Accept` header. EG from `Accept: application/*;version=36.0` to `Accept: application/*;version=37.0`.

VMware publish [differences between versions 36.0 and 37.0](https://developer.vmware.com/apis/1260/vmware-cloud-director/doc/diff/index.html). You can also use [my project to see the differences](https://github.com/ccouzens/vcloud-rest-openapi#differences-between-versions).

> Do you know that the 36.0 APIs can execute with different query?
Such as https://<api_host>/api/query?type=vApp in the old version.

That should work in both version 36.0 and 37.0.

> As you know, are there any APIs that can list all the vApps information in version 36.0 APIs? or are there any operations to access the vApps information by APIs.

Sorry, it has been too long since I last used VMware Cloud Director. And I mostly worked with orgs and VDCs. But I suspect your old APIs should continue to work with the new `Accept` HTTP header.

Hopefully that helps,

Chris

**Comment by chenchunwei24k** [comment]
Hi @ccouzens 

Thank you for helping!
This information helps me solve the current problem.
Very much appreciate your help!

Best Regards,
Andy
